ert_shared/status/tracker/evaluator.py

```diff
@@ -115,20 +124,66 @@ def _drain_monitor(self):
         drainer_logger.debug("tasks complete")
 
     def track(self):
-        while True:
-            event = self._work_queue.get()
-            if event is EvaluatorTracker.DONE:
+        done: bool = False
+        while not done:
+            events: List[CloudEvent] = []
+
+            while True:
                 try:
-                    yield EndEvent(
-                        failed=self._model.hasRunFailed(),
-                        failed_msg=self._model.getFailMessage(),
-                    )
-                except GeneratorExit:
-                    # consumers may exit at this point, make sure the last
-                    # task is marked as done
-                    self._work_queue.task_done()
-                break
-            elif event["type"] == ids.EVTYPE_EE_SNAPSHOT:
+                    event: CloudEvent = self._work_queue.get_nowait()
+                    if event is EvaluatorTracker.DONE:
+                        done = True
+                        break
+                    events.append(event)
+                except queue.Empty:
+                    break
+
+            if events:
+                yield from self._batch(events)
+            time.sleep(self._general_interval)
```

Comment:
Why do we need to sleep here? and what is the expected value for the `_general_interval` ? 1 or 5?

Reply:
It varies, but yes, as we discussed without the sleep, you might end up with batches of size 1, defeating the purpose of this PR, and most like not fixing the GUI lag.